### PR TITLE
Fix 557 suppress html5lib sanitizer deprecation warnings

### PR DESCRIPTION
**Pull request by g-k**
fixes: #557 

r? @willkg if you get a minute

### CODE REVIEW

File: bleach/__init__.py [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
+# ignore html5lib deprecation warnings to use bleach; we are bleach
+# apply before we import submodules that import html5lib
+warnings.filterwarnings(
+    "ignore",
+    message="html5lib's sanitizer is deprecated",
+    category=DeprecationWarning,
+    module="bleach._vendor.html5lib",
+)
+
+from bleach.linkifier import (  # noqa: E402 module level import not at top of file
```
Comment (willkg) [inline bleach/__init__.py:17]: E402 is so annoying. I can't think of a case where it's helped me. Might be worth excluding in the configuration.

**Comment by willkg** [comment]
It looks like this covers two cases:

1. `import bleach`
2. `from bleach import sanitizer`

Is it possible to put the warning-ignoring code at the top of `bleach/html5lib_shim.py` instead? That way all the html5lib-related things are in the same place which simplifies maintenance.

**Comment by g-k** [comment]
I moved the warning ignoring code from `__init__.py` to `bleach/html5lib_shim.py` which covers the warning in html5lib init https://github.com/mozilla/bleach/blob/b54c78e084b41be8fda8483f72740710377edd26/bleach/_vendor/html5lib/filters/sanitizer.py#L29

There's another in the `html5lib.filters.sanitizer::Filter.__init__` https://github.com/mozilla/bleach/blob/b54c78e084b41be8fda8483f72740710377edd26/bleach/_vendor/html5lib/filters/sanitizer.py#L771 

`from bleach import clean; clean('foo')` raises the warning even when the init warnings filter is in place, so I think both are necessary

File: bleach/html5lib_shim.py [hunk 2]
```diff
@@ -12,26 +12,51 @@
...
 
-# filter out html5lib deprecation warnings to use bleach
-warnings.simplefilter("ignore", category=DeprecationWarning)
+# ignore html5lib deprecation warnings to use bleach; we are bleach
+# apply before we import submodules that import html5lib
+warnings.filterwarnings(
+    "ignore",
+    message="html5lib's sanitizer is deprecated",
+    category=DeprecationWarning,
+    module="bleach._vendor.html5lib",
```
Comment (willkg) [inline bleach/html5lib_shim.py:21]: If you nix this `module=` bit here, then will this apply to the module and the Sanitizer `__init__`?

**Comment by g-k** [comment]
ok so, the `test_clean.py` failure is from:

```
» PYTHONDONTWRITEBYTECODE=1 pytest --cache-clear -x -v tests/test_clean.py
========================================================================== test session starts ==========================================================================
platform linux -- Python 3.8.2, pytest-6.0.2, py-1.9.0, pluggy-0.13.1 -- /home/gguthe/bleach/venv/bin/python
cachedir: .pytest_cache                                        
rootdir: /home/gguthe/bleach, configfile: setup.cfg        
plugins: wholenodeid-0.2                                                                                                                                                 collected 183 items                                                                                                                                                     
                                                             
tests/test_clean.py::test_clean_idempotent FAILED                                                                                                                 [  0%]
                                                                                                                                                                         
=============================================================================== FAILURES ================================================================================
______________________________________________________________ tests/test_clean.py::test_clean_idempotent _______________________________________________________________
                                                                                                                                                                         
    def test_clean_idempotent():                                               
        """Make sure that applying the filter twice doesn't change anything."""
        dirty = "<span>invalid & </span> < extra http://link.com<em>"
>       assert clean(clean(dirty)) == clean(dirty)

tests/test_clean.py:16:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
bleach/__init__.py:90: in clean
    return cleaner.clean(text)
bleach/sanitizer.py:177: in clean
    filtered = BleachSanitizerFilter(
bleach/sanitizer.py:288: in __init__
    return super(BleachSanitizerFilter, self).__init__(source, **kwargs)
```

File: bleach/html5lib_shim.py (hunk 2 above)
Comment (g-k) [reply bleach/html5lib_shim.py:21]: no that still fails

**Comment by g-k** [comment]
however, a separate test with:

```python
    from bleach import clean
    # from bleach.html5lib_shim import Filter                                                                                                                            
    # from bleach.sanitizer import Cleaner                                                                                                                               
    # from bleach._vendor.html5lib.constants import rcdataElements                                                                                                       
    clean('foo')
```

does not fail

**Comment by pllim** [comment]
Hello. Thank you for the quick fix! In which release should I expect this patch to be in?

**Comment by g-k** [comment]
It's out in 3.2.1 https://pypi.org/project/bleach/3.2.1/

Thanks for reporting the bug!

**Comment by pllim** [comment]
Amazing. Thank you!

**Pull request merged**